device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1218,6 +1230,12 @@ private void reconnect(TransportException transportException)
                 && transportException != null
                 && transportException.isRetryable())
         {
+            if (this.isClosing)
```

Comment:
This check occurs in the reconnection loop, so if ```close()``` is called while this reconnection is already happening, it will block until the latest reconnection attempt completes and the thread exits before the ```close()``` call continues.